op-chain-ops/genesis/db_migration.go

```diff
@@ -19,96 +22,142 @@ import (
 	withdrawals, err := migrationData.ToWithdrawals()
 	if err != nil {
-		return fmt.Errorf("cannot serialize withdrawals: %w", err)
+		return nil, fmt.Errorf("cannot serialize withdrawals: %w", err)
 	}
 
 	if err := CheckWithdrawals(db, withdrawals); err != nil {
-		return fmt.Errorf("withdrawals mismatch: %w", err)
+		return nil, fmt.Errorf("withdrawals mismatch: %w", err)
 	}
 
 	// Now start the migration
 	if err := SetL2Proxies(db); err != nil {
-		return fmt.Errorf("cannot set L2Proxies: %w", err)
+		return nil, fmt.Errorf("cannot set L2Proxies: %w", err)
 	}
 
 	storage, err := NewL2StorageConfig(config, l1Block, l2Addrs)
 	if err != nil {
-		return fmt.Errorf("cannot create storage config: %w", err)
+		return nil, fmt.Errorf("cannot create storage config: %w", err)
 	}
 
 	immutable, err := NewL2ImmutableConfig(config, l1Block, l2Addrs)
 	if err != nil {
-		return fmt.Errorf("cannot create immutable config: %w", err)
+		return nil, fmt.Errorf("cannot create immutable config: %w", err)
 	}
 
 	if err := SetImplementations(db, storage, immutable); err != nil {
-		return fmt.Errorf("cannot set implementations: %w", err)
+		return nil, fmt.Errorf("cannot set implementations: %w", err)
 	}
 
 	err = crossdomain.MigrateWithdrawals(withdrawals, db, &l2Addrs.L1CrossDomainMessengerProxy, &l2Addrs.L1StandardBridgeProxy)
 	if err != nil {
-		return fmt.Errorf("cannot migrate withdrawals: %w", err)
+		return nil, fmt.Errorf("cannot migrate withdrawals: %w", err)
 	}
 
 	addrs := migrationData.Addresses()
-	if err := ether.MigrateLegacyETH(ldb, addrs, migrationData.OvmAllowances, int(config.L1ChainID)); err != nil {
-		return fmt.Errorf("cannot migrate legacy eth: %w", err)
-	}
-
-	if !commit {
-		return nil
-	}
-
-	root, err := db.Commit(true)
+	newRoot, err := ether.MigrateLegacyETH(ldb, addrs, migrationData.OvmAllowances, int(config.L1ChainID), commit)
 	if err != nil {
-		return fmt.Errorf("cannot commit state db: %w", err)
+		return nil, fmt.Errorf("cannot migrate legacy eth: %w", err)
 	}
 
 	// Create the bedrock transition block
 	bedrockHeader := &types.Header{
 		ParentHash:  header.Hash(),
 		UncleHash:   types.EmptyUncleHash,
 		Coinbase:    config.L2GenesisBlockCoinbase,
-		Root:        root,
+		Root:        newRoot,
 		TxHash:      types.EmptyRootHash,
 		ReceiptHash: types.EmptyRootHash,
 		Bloom:       types.Bloom{},
-		Difficulty:  (*big.Int)(config.L2GenesisBlockDifficulty),
+		Difficulty:  common.Big0,
 		Number:      new(big.Int).Add(header.Number, common.Big1),
 		GasLimit:    (uint64)(config.L2GenesisBlockGasLimit),
-		GasUsed:     (uint64)(config.L2GenesisBlockGasUsed),
+		GasUsed:     0,
 		Time:        uint64(config.L2OutputOracleStartingTimestamp),
-		Extra:       config.L2GenesisBlockExtraData,
-		MixDigest:   config.L2GenesisBlockMixHash,
-		Nonce:       types.EncodeNonce((uint64)(config.L1GenesisBlockNonce)),
+		Extra:       []byte("BEDROCK"),
+		MixDigest:   common.Hash{},
+		Nonce:       types.BlockNonce{},
```

Comment:
This must be empty otherwise the opnode fails block validation.

Reply:
Both the nonce and mix digest?

Reply:
Longer term we will want to remove anything from the config that is not required

Reply:
zero block nonce is the right thing yes, we don't want any PoW legacy to be non-zero. The uncles root should be set to the empty-uncles hash. See https://eips.ethereum.org/EIPS/eip-3675 for more details.

Reply:
And extra-data should be no longer than 32 bytes, but may be shorter. The `BEDROCK` value looks good to me.